tools/jsii-compliance/report.ts

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env npx ts-node
+
+import * as fs from 'fs';
+import * as path from 'path';
+
+import * as schema from './schema';
+import { suite } from './suite';
+
+// eslint-disable-next-line @typescript-eslint/no-require-imports,@typescript-eslint/no-var-requires
+const tablemark = require('tablemark');
+
+const SUCCESS = '✅';
+const FAILURE = '❌';
+const NA = 'N/A';
+
+/**
+ * Determine whether or not a test case should be excluded from a specific language.
+ * An exclusion happens either when the entire language binding is excluded, or a specific test case.
+ *
+ * @param test the test.
+ * @param suite the suite.
+ * @param language the language.
+ */
+function isExcluded(
+  testCase: schema.TestCase,
+  suite: schema.Suite,
+  language: string,
+) {
+  const testExcluded =
+    testCase.exclusions && testCase.exclusions[language] ? true : false;
+  const bindingExcluded =
+    suite.exclusions && suite.exclusions[language] ? true : false;
+  return testExcluded || bindingExcluded;
+}
+
+/**
+ * Determines the status of a specific test case with respect to a specific language.
+ *
+ * @param testCase the test case.
+ * @param language test language.
+ * @param reports the reports collected from all language bindings.
+ */
+function determineTestStatus(
+  testCase: schema.TestCase,
+  language: string,
+  reports: Record<string, schema.Report>,
+) {
+  const report = reports[language];
+  const testResult = report
+    ? report[normalizeTestName(testCase.name)]
+    : undefined;
+
+  if (!testResult) return FAILURE;
+  return testResult.status === 'success' ? SUCCESS : FAILURE;
+}
+
+/**
+ * Given a test name, normalize it so it can be compared across different language bindings.
+ *
+ * @param testName the test name.
+ */
+function normalizeTestName(testName: string): string {
+  return testName.toUpperCase();
+}
+
+/**
+ * Given a compliance report, normalize its test names so they are comparable to the
+ * tests defined in the suite.
+ *
+ * @param report report
+ */
+function normalizeReport(report: schema.Report): schema.Report {
+  const normalized: schema.Report = {};
+  Object.keys(report).forEach((k) => {
+    normalized[normalizeTestName(k)] = report[k];
+  });
```

Comment:
I kinda find the below more expressive:

```suggestion
  for (const [key, value] of Object.entries(report)) {
    normalized[normalizeTestName(key)] = value;
  }
```

Reply:
Yeah, better for debugging too. 👍